design/one-pager-easy-stacks.md

```diff
@@ -0,0 +1,294 @@
+2. Install a AWS configuration stack via:
+```yaml
+# exact apiVersion is TBD.
+apiVersion: stacks.crossplane.io/v1alpha1
+kind: ClusterStackInstall
+metadata:
+  name: easy-aws
+  namespace: crossplane-system
+spec:
+  package: "crossplane/easy-aws:latest"
+```
+
+Now, I want an environment where all my database instances and kubernetes clusters
+are connected to each other in a private VPC, which what this specific configuration
+stack does. Create the following:
+
+```yaml
+# exact apiVersion is TBD.
+apiVersion: aws.configurationstacks.crossplane.io/v1alpha1
+kind: EasyAWS
+metadata:
+  name: project-future
+  namespace: dev
+spec:
+  providerRef:
+    name: aws-provider
+```
+
+Then I wait for status to become ready. After it's done, all resources are deployed.
+```yaml
+# exact apiVersion is TBD.
+apiVersion: aws.configurationstacks.crossplane.io/v1alpha1
+kind: EasyAWS
+metadata:
+  name: project-future
+  namespace: dev
+spec:
+  providerRef:
+    name: aws-provider
+status:
+  state: ready
+```
+
+An example deployed resource would be:
+```yaml
+apiVersion: database.aws.crossplane.io/v1beta1
+kind: RDSInstanceClass
+metadata:
+  name: dev-project-future-mysql
+  labels:
+    "aws.configurationstacks.crossplane.io/name": project-future
+    "aws.configurationstacks.crossplane.io/namespace": dev
+    "aws.configurationstacks.crossplane.io/uid": 1ca16960-973b-4d58-a6fa-5696638ec631
+specTemplate:
+  writeConnectionSecretsToNamespace: crossplane-system
+  forProvider:
+    dbInstanceClass: db.t2.small
+    masterUsername: root
+    vpcSecurityGroupIDRefs:
+      - name: dev-project-future-rds-security-group
+    dbSubnetGroupNameRef:
+      name: dev-project-future-dbsubnetgroup
+    allocatedStorage: 20
+    engine: mysql
+    skipFinalSnapshotBeforeDeletion: true
+  providerRef:
+    name: aws-provider
+  reclaimPolicy: Delete
+```
+
+Other people in different namespaces can create their own instances of `EasyAWS`
+custom resource and have their own similar environment with different names.
+
+## Technical Implementation
+
+### Metadata of The Stack
+
+[Dependency list] has to be filled with each kind of the resource that the stack
+deploys, i.e. f the stack only deploys `RDSInstanceClass` resource, it should
+not require all AWS stack to be present.
+
+[UI annotations] should be present to make it easy for frontend software to process
+the stack.
+
+### Folder Structure
+
+All directories containing base YAML files should be in a top-level directory
+of the repository named `resources` so that it's easier for others to fork & change.
+
+Controller will scan this folder recursively to get all YAML files. So, the folder
+structure inside that `resources` directory won't matter to the controller but
+it's good to separate the folders cleanly so that it's easier to fork & change.
+
+There are mainly 2 types of resources that configuration stacks will usually
+need to deploy; resource classes and managed resources. Under `resources` folder
```

Comment:
Grammar nitpick: I would normally use a colon when giving an example or a list of examples, so `deploy: `. Semicolons (`;`) are for [connecting related clauses](https://en.wikipedia.org/wiki/Semicolon#Usage); often a semicolon can be used where a period would also work.